patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -1933,68 +1933,261 @@
  						}
++						SkipWallSnowConversion:
++						if (!convertTile)
++							goto SkipTileSnowConversion;
+ 						if ((TileID.Sets.Conversion.Grass[type] || TileID.Sets.Conversion.Sand[type] || TileID.Sets.Conversion.HardenedSand[type] || TileID.Sets.Conversion.Snow[type] || TileID.Sets.Conversion.Dirt[type]) && type != 147) {
+ 							TryKillingTreesAboveIfTheyWouldBecomeInvalid(k, l, 147);
+ 							tile.type = 147;
+@@ -39714,10 +_,13 @@
+ 							if (Main.netMode == 1)
+ 								NetMessage.SendData(17, -1, -1, null, 0, k, l);
+ 						}
++						SkipTileSnowConversion:
  						continue;
  					case 7:
 -						if (type < 0 || type > TileID.Count || wall < 0 || wall > WallID.Count)
 +						if (type < 0 || type >= TileLoader.TileCount || wall < 0 || wall >= WallLoader.WallCount)
  							continue;
++						if (!convertWall)
++							goto SkipWallForestConversion;
  						if ((WallID.Sets.Conversion.Stone[wall] || WallID.Sets.Conversion.Ice[wall] || WallID.Sets.Conversion.Sandstone[wall]) && wall != 1) {
  							tile.wall = 1;
+ 							SquareWallFrame(k, l);
+@@ -39748,6 +_,9 @@
+ 							SquareWallFrame(k, l);
+ 							NetMessage.SendTileSquare(-1, k, l);
+ 						}
++						SkipWallForestConversion:
++						if (!convertTile)
++							goto SkipTileForestConversion;
+ 						if ((TileID.Sets.Conversion.Stone[type] || TileID.Sets.Conversion.Ice[type] || TileID.Sets.Conversion.Sandstone[type]) && type != 1) {
+ 							TryKillingTreesAboveIfTheyWouldBecomeInvalid(k, l, 1);
+ 							tile.type = 1;
+@@ -39781,8 +_,12 @@
+ 							if (Main.netMode == 1)
+ 								NetMessage.SendData(17, -1, -1, null, 0, k, l);
+ 						}
++						SkipTileForestConversion:
+ 						continue;
+ 				}
++
++				if (!convertWall)
++					goto SkipWallPurityConversion;
+ 
+ 				if (tile.wall == 69 || tile.wall == 70 || tile.wall == 81) {
+ 					if ((double)l < Main.worldSurface) {
+@@ -39857,6 +_,10 @@
+ 					NetMessage.SendTileSquare(-1, k, l);
+ 				}
+ 
++				SkipWallPurityConversion:
++				if (!convertTile)
++					goto SkipTilePurityConversion;
++
+ 				if (tile.type == 492) {
+ 					TryKillingTreesAboveIfTheyWouldBecomeInvalid(k, l, 477);
+ 					tile.type = 477;
+@@ -39914,9 +_,52 @@
+ 					if (Main.netMode == 1)
+ 						NetMessage.SendData(17, -1, -1, null, 0, k, l);
+ 				}
++
++				SkipTilePurityConversion:
++				;
+ 			}
+ 		}
+ 	}
++
++	/// <summary>
++	/// Converts the single tile at the given coordinate into a specified new tile type<br/>
++	/// Automatically handles tile framing and multiplayer syncing.
++	/// </summary>
++	/// <param name="i">The X coordinate of the target tile.</param>
++	/// <param name="j">The Y coordinate of the target tile.</param>
++	/// <param name="newType">The new type to convert the tile into</param>
++	/// <param name="tryBreakTrees">Should the conversion try to break trees above the converted tile if the new type is invalid for the tree</param>
++	public static void ConvertTile(int i, int j, int newType, bool tryBreakTrees = false)
```

Comment:
Is this method recommended for use with custom clentaminator likes, or only in `RandomUpdate`?